### PR TITLE
Mapbox info control

### PR DESCRIPTION
**Pull request by tristen**
This commit adds a new info control in reference to #485 
- [x] Add and remove attribution on layerAdd and layerRemove like [in L.Control.Attribution](https://github.com/Leaflet/Leaflet/blob/master/src/control/Control.Attribution.js#L27-L29)
- [x] IE test

I stuck with an _info_ icon as this felt visually stronger as an icon as is more diverse than a &copy; I've also placed this control in the bottom left corner (as opposed to the bottom right) as it created some awkwardness when a legend was just above it.

Here's a little animation of it. The control appends any attribution content passed in to `L.mapbox.infoControl.addInfo()` and adds an "Improve this Map" link by default. There's also a supported removeInfo method. 

attribution

### CODE REVIEW

**Comment by incanus** [comment]
Was gonna say: in addition to any other reasons, the edit in OSM link could be really tricky on mobile web devices, where I don't think iD is meant to tread just yet...

**Comment by tmcw** [comment]
From meatspace: we are re-adding edit in osm, making it off by default, and showing it when (requested && !mobile)

**Comment by tristen** [comment]
@incanus right - I'm conditioning the display of this on mobile with `!L.Browser.mobile`

File: src/info_control.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+var InfoControl = L.Control.extend({
+    options: {
+        position: 'bottomleft',
+        sanitizer: require('sanitize-caja'),
+        editinosm: false
```
Comment (jfirebaugh) [inline src/info_control.js]: Options names are camel case, suggest `editLink`.

Comment (tristen) [inline src/info_control.js]: Sounds good

File: src/info_control.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        return this;
+    },
+
+    _osmlink: function() {
+        var center = this._map.getCenter();
+        var z = this._map.getZoom();
+        window.open('http://www.openstreetmap.org/edit?' + 'zoom=' + z +
+        '&editor=id' + '&lat=' + center.lat + '&lon=' + center.lng);
```
Comment (jfirebaugh) [inline src/info_control.js]: Don't include `editor=id` here, the default is to use an appropriate editor automatically (iD for most users, preferred editor for users who have set one, P2 for IE users).

File: src/map.js [hunk 3]
```diff
@@ -15,7 +16,9 @@ var Map = L.Map.extend({
         markerLayer: {},
         gridLayer: {},
         legendControl: {},
-        gridControl: {}
+        gridControl: {},
+        infoControl: {},
+        attributionControl: false
```
Comment (jfirebaugh) [inline src/map.js]: Should L.mapbox.map _not_ create an infoControl by default if someone explicitly passes `{attributionControl: true}`?

**Pull request merged**